src/backend/mi2/mi2.ts

```diff
@@ -373,6 +405,17 @@ export class MI2 extends EventEmitter implements IBackend {
 										switch (reason) {
 											case "breakpoint-hit":
 												this.emit("breakpoint", parsed);
+												this.logMsg.forEach((brk, index)=>{
+													if(parsed.outOfBandRecord[0].output[0][1] == "breakpoint-hit" && parsed.outOfBandRecord[0].output[2][1] == brk.id){
+														this.logMsgNum = brk.id;
+														this.logMsgVar = brk?.logMessage;
+														const matches = this.logMsgVar.match(logReplaceTest);
+														const count = matches ? matches.length : 0;
+														this.logMsgRplNum = count;
+														this.logMsgVarProcess = this.logMsgVar.replace(logReplaceTest, "placeHolderForVariable");
```

Comment:
this seems rather random and temporary.

Can you move the functionality here + the new variables into a helper class, so it's easier to reason about the logic?